app/src/androidTest/java/org/dhis2/common/idlingresources/MapIdlingResource.kt

```diff
@@ -0,0 +1,40 @@
+package org.dhis2.common.idlingresources
+
+import androidx.test.espresso.IdlingResource
+import androidx.test.rule.ActivityTestRule
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
+import org.dhis2.R
+
+class MapIdlingResource(
+    activityTestRule: ActivityTestRule<*>
+) : IdlingResource, OnMapReadyCallback {
+
+    var map: MapboxMap? = null
+    private var resourceCallback: IdlingResource.ResourceCallback? = null
+
+    init {
+        try {
+            val mapView = activityTestRule.activity.findViewById<MapView>(R.id.mapView)
+            mapView.getMapAsync(this)
```

Comment:
@ferdyrod do you think we need to wait for styles?

Reply:
We can wait for whatever we need. In this PR it was just to get the map working with an IdlingResource. I believe we shoudl have a separated module for testing the map features.

However, in my testing I was getting the styles with its layers information in the test.